### PR TITLE
fix: deep copy items in arrays better

### LINKED ISSUE #434: :bug: i18n not populating
i18n isn't populating.

thoughts:
- bad cache key needs to include value from Accept-Language?
- i18n isn't coming in from warframe-items
- i18n isn't generating on warframe

### PR DESCRIPTION
**Pull request by TobiTenno**
### What did you fix?
closed #434 

---

### Reproduction steps
- add deep copy implementation to deep copy items. for whatever reason, previous attempts weren't covering it. added commensurate tests
- ci: more coverage

TODO:
- [x] check static data language selection

---

### Considerations
- Does this contain a new dependency? **[No]**
- Does this introduce opinionated data formatting or manual data entry? **[No]**
- Does this pr include updated data files in a separate commit that can be reverted for a clean code-only pr? **[No]**
- Have I run the linter? **[Yes]**
- Is is a bug fix, feature request, or enhancement? **[Bug Fix]**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -79,6 +79,6 @@ jobs:
       - name: Update server
         run: ssh prod 'source ~/.zshrc && cd ~/warframe-status && git checkout -- . && git fetch --all && git reset --hard origin/main'
       - name: Update dependencies
-        run: ssh prod 'source ~/.zshrc && cd ~/warframe-status && rm -rf node_modules && npm ci'
+        run: ssh prod 'source ~/.zshrc && cd ~/warframe-status && rm -rf node_modules && nvm install && nvm use && npm ci'
```
Comment (TobiTenno) [inline .github/workflows/release.yml:82]: Added this to make updates smoother when updating node versions

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-lts/gallium
+lts/fermium
```
Comment (TobiTenno) [inline .nvmrc:1]: Node 14 instead of 16

File: src/controllers/index.js [hunk 3]
```diff
@@ -4,23 +4,34 @@ const router = require('express').Router();
...
+  /* istanbul ignore if */
+  if (!platforms.includes(req.platform)) req.platform = 'pc';
+
+  req.language = (req.header('Accept-Language') || 'en').substring(0, 2).toLowerCase();
+  req.language = (req.query.language || req.language).substring(0, 2);
+  if (!(req.language && languages.includes(req.language))) req.language = 'en';
+
+  next();
+});
+
```
Comment (TobiTenno) [inline src/controllers/index.js:27]: Moved this here to add some common parameter checking so it didn't need to get copied

File: src/spec/items.spec.js [hunk 4]
```diff
@@ -76,14 +76,60 @@ describe('items', () => {
...
+    res.body[0].i18n.should.have.property('zh');
+    res.should.have.header('Content-Language', 'zh');
+
+    res = await chai.request(server)
+      .get('/items/search/excalibur%20umbra?language=zh&only=i18n,name');
+    res.should.have.status(200);
+    res.body.should.be.an('array');
+    res.body.length.should.be.greaterThan(1);
+    res.body[0].i18n.should.have.property('zh');
+    res.body[0].i18n.should.not.have.property('it');
```
Comment (TobiTenno) [inline src/spec/items.spec.js:96]: Added a test for the scenario we were seeing where using a different non-english locale after another has been used causes the i18n to be empty

**Review by SlayerOrnstein — APPROVED** [review]

**Pull request merged**